### PR TITLE
Update math classes

### PR DESCRIPTION
**Pull request by larsr**
###### Motivation for this change

A new version of math-classes that works with the default coq version in nix is available.

###### Things done

- [x] added checksum for the new release
- [x] tested to compile it for a few coq versions, i.e 8.6, 8.10, 8.11, 8.12, 8.13 

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Zimmi48** [comment]
Test started at https://github.com/coq-community/coq-nix-toolbox/pull/51 as per https://github.com/coq-community/coq-nix-toolbox#testing-coqpackages-updates-in-nixpkgs

File: pkgs/development/coq-modules/math-classes/default.nix [hunk 1]
```diff
@@ -4,8 +4,10 @@ with lib; mkCoqDerivation {
 
   pname = "math-classes";
   inherit version;
-  defaultVersion = if versions.range "8.6" "8.12" coq.coq-version then "8.12.0" else null;
+  defaultVersion = if versions.range "8.11" "8.13" coq.coq-version then "8.13.0" else
+                   if versions.range "8.6"  "8.10" coq.coq-version then "8.12.0" else null;
```
Comment (Zimmi48) [inline pkgs/development/coq-modules/math-classes/default.nix:8]: Note that you could also write this with a switch like this:

```suggestion
  defaultVersion = with versions; switch coq.coq-version [
    { case = range "8.11" "8.13"; out = "8.13.0"; }
    { case = range "8.6"  "8.10"; out = "8.12.0"; }
  ] null;
```

Up to you as the two mean the same thing.

**Review by langston-barrett — APPROVED** [review]

**Comment by larsr** [comment]
@Zimmi48 The unfortunate need to keep using the old math-class version for compilers <= 8.10 is because of the change to using `lia` instead of `omega` [here](https://github.com/coq-community/math-classes/blob/b94b24aa387abb90740acf7cc24286bf27d0fb50/theory/monoid_normalization.v#L76). Perhaps one should use something else than `lia` to keep backwards compatibility?  Or is it better to forge ahead and only support old coq versions with old library versions?

**Comment by Zimmi48** [comment]
>  Or is it better to forge ahead and only support old coq versions with old library versions?

Yeah, that's my point of view. Library maintainers can only do so much and support a reasonable range of Coq version. Beyond that, it becomes too difficult.

File: pkgs/development/coq-modules/math-classes/default.nix (hunk 1 above)
Comment (larsr) [reply pkgs/development/coq-modules/math-classes/default.nix:8]: If you don't mind, I'll keep the if formulation for now. Hopefully the condition can be returned to its previous simple  form some day if https://github.com/coq-community/math-classes/pull/102 is approved.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129766` at fdfbe869 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>coqPackages.math-classes</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/build-support/coq/default.nix:75:3:
```
   |
75 |   meta = ({ platforms = coq.meta.platforms; } //
   |   ^
```
</li>
  </ul>
</details>

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
@vbgl Would you merge this PR?

**Comment by Zimmi48** [comment]
Thanks!

**Pull request merged**